odoo/tests/common.py

```diff
@@ -1378,21 +1378,43 @@ def replacer(m):
         return replacer
 
 
+class Opener(requests.Session):
+    """
+    Flushes and clears the current transaction when starting a request.
+
+    This is likely necessary when we make a request to the server, as the
+    request is made with a test cursor, which uses a different cache than this
+    transaction.
+    """
+    def __init__(self, cr: BaseCursor):
+        super().__init__()
+        self.cr = cr
+
+    def request(self, *args, **kwargs):
+        self.cr.flush(); self.cr.clear()
+        return super().request(*args, **kwargs)
+
+
+class Transport(xmlrpclib.Transport):
+    """ see :class:`Opener` """
+    def __init__(self, cr: BaseCursor):
+        self.cr = cr
+        super().__init__()
+
+    def request(self, *args, **kwargs):
+        self.cr.flush(); self.cr.clear()
+        return super().request(*args, **kwargs)
+
+
 class HttpCase(TransactionCase):
     """ Transactional HTTP TestCase with url_open and Chrome headless helpers. """
     registry_test_mode = True
     browser = None
     browser_size = '1366x768'
 
-    def __init__(self, methodName='runTest'):
-        super().__init__(methodName)
-        # v8 api with correct xmlrpc exception handling.
-        self.xmlrpc_url = url_8 = 'http://%s:%d/xmlrpc/2/' % (HOST, odoo.tools.config['http_port'])
-        self.xmlrpc_common = xmlrpclib.ServerProxy(url_8 + 'common')
-        self.xmlrpc_db = xmlrpclib.ServerProxy(url_8 + 'db')
-        self.xmlrpc_object = xmlrpclib.ServerProxy(url_8 + 'object')
-        cls = type(self)
-        cls._logger = logging.getLogger('%s.%s' % (cls.__module__, cls.__name__))
+    env: api.Environment = None
+    cr: BaseCursor = None
+    _logger: logging.Logger = None
```

Comment:
Tempted by pseudo-static typing?  Looks overkill to me...  But mostly confusing for `env` and `cr`, which are actually set by `TransactionCase`.

Reply:
> Tempted by pseudo-static typing?

I just declared the attributes because otherwise pycharm didn't understand where they come from (since they're defined in a method which it doesn't like). While at it it was no harder to actually type them.

Reply:
Don't get me wrong.  I am not against those annotations, but I think that `env` and `cr` should be annotated on `TransactionCase`.  This would feel more consistent.

Reply:
Sure, was just explaning why it is they landed here.